### ISSUE TITLE
Using multicost on different labels

### ISSUE THREAD

**Issue by mtamburrano** [NervanaSystems/neon#159]
Is it possible to build a net and use multiple cost on different labels in the same dataset?
Let's say I have a dataset X and I want to build a net with a `SumSquared` cost to localize some feature and a `Softmax` layer with a `CrossEntropyMulti` to classify the image, so my labels should be something like Y=[masks, classes].
I see it is possible to set different branches on the net and use different costs function, but I don't know how to differentiate the labels and feed the right cost with the right label, because it seems they are managed by the mlp class using an iterable on the dataset.
Is there a workaround?

**Comment by bhack** [comment]
:100:

**Comment by hanlin-nervana** [comment]
To provision multiple labels to the model, we would recommend implementing a custom iterator that yields for each minibatch a tuple of `(X, (Y1, Y2))`. The two target labels will be matched against the output branches of your network using the associated costs.

See:
https://github.com/NervanaSystems/neon/tree/master/examples/fast-rcnn
https://github.com/NervanaSystems/neon_course/blob/master/03%20Writing%20your%20own%20dataset.ipynb

**Comment by hanlin-nervana** [comment]
Closing this issue, please reopen if you have further questions, thanks!
